x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/RuntimeUtils.java

```diff
@@ -173,7 +171,7 @@ public static HitExtractor createExtractor(FieldExtraction ref, EqlConfiguration
     }
 
     public static SearchRequest prepareRequest(SearchSourceBuilder source, boolean includeFrozen, String... indices) {
-        SearchRequest searchRequest = new SearchRequest(SWITCH_TO_MULTI_VALUE_FIELDS_VERSION);
```

Comment:
@astefan Is this change ok? There's no chance this request could go to a pre-7.17 node/cluster?

Reply:
@thecoop this min version requirement has been introduced for bwc support. This means that any scenario where a rolling upgrade is possible that involves versions pre-7.15.0 with version post-7.15.0 will use this mechanism. In other words, if at this moment we don't support nodes having version 8.12 (where the change in this PR goes to) and nodes having version <7.15.0 (what `SWITCH_TO_MULTI_VALUE_FIELDS_VERSION` is) **in the same cluster** (a rolling upgrade scenario) then it should be ok to make this change.

Reply:
Great - v8 can only ever connect to 7.17, so this change is ok

Reply:
[Our current documentation (for 8.11)](https://www.elastic.co/guide/en/elasticsearch/reference/8.11/setup-upgrade.html) states this:
> To upgrade to 8.11.1 from 7.16 or an earlier version, you must first upgrade to 7.17, even if you opt to do a full-cluster restart instead of a rolling upgrade. This enables you to use the Upgrade Assistant to identify and resolve issues, reindex indices created before 7.0, and then perform a rolling upgrade. You must resolve all critical issues before proceeding with the upgrade. For instructions, refer to [Prepare to upgrade from 7.x](https://www.elastic.co/guide/en/elastic-stack/8.11/upgrading-elastic-stack.html#prepare-to-upgrade).

This means that your change should be ok. It is not possible to do any kind of upgrade from 7.15 to 8.12 where these two versions exists in the same cluster. Nodes must be upgrade to 7.17 and then to 8.11.